ambry-router/src/main/java/com.github.ambry.router/PutManager.java

```diff
@@ -221,6 +246,12 @@ boolean isOpen() {
    */
   void close() {
     if (isOpen.compareAndSet(true, false)) {
+      synchronized (chunkFillerSynchronizer) {
+        if (isChunkFillerThreadAsleep) {
+          chunkFillerThreadSleepDisabled = true;
```

Comment:
not sure why you need this new variable. Won't checking `isOpen` be enough ? (both variables seem to carry the same piece of information).